WordPress/src/test/java/org/wordpress/android/push/NotificationHelperTest.kt

```diff
@@ -0,0 +1,114 @@
+
+private const val PUSH_TYPE_NOTE_A_COMMENT = "NotAComment"
+
+@ExperimentalCoroutinesApi
+class NotificationHelperTest : BaseUnitTest() {
+    private val systemNotificationsTracker: SystemNotificationsTracker = mock()
+    private val gcmMessageHandler: GCMMessageHandler = mock()
+    private val notificationsUtilsWrapper = mock<NotificationsUtilsWrapper>()
+    private val notificationHelper =
+        GCMMessageHandler.NotificationHelper(gcmMessageHandler, systemNotificationsTracker, notificationsUtilsWrapper)
+
+    @Test
+    fun `WHEN a PN that is a comment has a message argument THEN then the message is used as title`() {
+        val expectedTitle = "expectedTitle"
+        val defaultTitle = "defaultTitle"
+        val mockedBundle = mock<Bundle>()
+        whenever(mockedBundle.getString(PUSH_ARG_MSG)).thenReturn(StringEscapeUtils.escapeHtml4(expectedTitle))
+        val title = notificationHelper.getNotificationTitle(mockedBundle, PUSH_TYPE_COMMENT, defaultTitle)
+        assertEquals(expectedTitle, title)
+    }
+
+    @Test
+    fun `WHEN a PN that is not a comment contains a title argument THEN this title is used`() {
+        val expectedTitle = "expectedTitle"
+        val defaultTitle = "defaultTitle"
+        val mockedBundle = mock<Bundle>()
+        whenever(mockedBundle.getString(PUSH_ARG_TITLE)).thenReturn(StringEscapeUtils.escapeHtml4(expectedTitle))
+        val title = notificationHelper.getNotificationTitle(mockedBundle, PUSH_TYPE_NOTE_A_COMMENT, defaultTitle)
+        assertEquals(expectedTitle, title)
+    }
+
+    @Test
+    fun `WHEN a PN that is not a comment does not contain a title argument THEN the default title is used`() {
+        val defaultTitle = "defaultTitle"
+        val mockedBundle = mock<Bundle>()
+        whenever(mockedBundle.getString(PUSH_ARG_TITLE)).thenReturn(null)
+        val title = notificationHelper.getNotificationTitle(mockedBundle, PUSH_TYPE_NOTE_A_COMMENT, defaultTitle)
+        assertEquals(defaultTitle, title)
+    }
+
+    @Test
+    fun `WHEN a PN that is not a comment contains a message argument THEN the message is used`() {
+        val expectedMessage = "expectedMessage"
+        val mockedBundle = mock<Bundle>()
+        whenever(mockedBundle.getString(PUSH_ARG_MSG)).thenReturn(StringEscapeUtils.escapeHtml4(expectedMessage))
+        val message = notificationHelper.getNotificationMessage(mockedBundle, PUSH_TYPE_NOTE_A_COMMENT)
+        assertEquals(expectedMessage, message)
+    }
+
+    @Test
+    fun `WHEN a PN that is not a comment does not contain a message argument THEN an empty message is used`() {
+        val mockedBundle = mock<Bundle>()
+        whenever(mockedBundle.getString(PUSH_ARG_MSG)).thenReturn(null)
+        val message = notificationHelper.getNotificationMessage(mockedBundle, PUSH_TYPE_NOTE_A_COMMENT)
+        assertEquals("", message)
+    }
+
+    @Test
+    fun `WHEN a PN that is a comment has a comment payload THEN the comment is used as a message`() {
```

Comment:
Not suggesting a change here, but curious about this the term "comment payload" in the test name (in this and following tests). Is that the same meaning as "comment subject", as is retrieved with `mockedNote.commentSubject`?

Reply:
Good question @mkevins 👍 
I felt that subject was misleading in this case. We get the comment content from the 2nd position of the `subject` array of the json that is passed in the PN `note_full_data` field. I think the `commentSubject` is a bit misleading in this case since the subject/title is passed in the 1st position on the `subject` array and this function uses the 2nd. I restrained from renaming it since it is used in other place and notification types too.
```
{
  ...
  "type": "comment",
  ...
  "subject": [
    {
      "text": "Antonis Lilis commented on Long title to test Push Notifications rendering Long title to test Push Notifications rendering"
    },
    {
      "text": "This is a dummy comment to test the push notifications functionality. This is a dummy comment to test the push notifications functionality. This is a dummy comment to …\n"
    }
  ],
  ...
}
```

Reply:
> The commentSubject is a bit misleading in this case since the subject/title is passed in the 1st position on the subject array and [this function uses the 2nd](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/models/Note.java#L222). I restrained from renaming it since it is used in other place and notification types too.

Maybe we can use a better naming in the `Note.java` "wrapper" we were discussing to create as an abstraction layer.

Reply:
Oh, ok, that makes more sense now. Thanks for the explanation!